### PR TITLE
Fix for Resource addition and updated unit tests.

### PR DESCRIPTION
**Pull request by yogpan01**
Adding more fixes related with Object operations.

### CODE REVIEW

**Comment by yogpan01** [comment]
@teetak01 @anttiylitokola Please review and try to run TLV test cases related with Object, Object Instance, Resource, ResourceInstances

**Comment by yogpan01** [comment]
I still have to update unit test today to improve the coverage.

File: source/m2mnsdlinterface.cpp [hunk 1]
```diff
@@ -471,9 +471,9 @@ uint8_t M2MNsdlInterface::received_from_server_callback(struct nsdl_s * nsdl_han
                                     M2MObject* object = (M2MObject*)base;
                                     M2MObjectInstance *obj_instance = object->create_object_instance(instance_id);
                                     if(obj_instance) {
-                                        obj_instance->set_operation(M2MBase::GET_PUT_ALLOWED);
-                                    }
-                                    coap_response = object->handle_post_request(_nsdl_handle,coap_header,this);
+                                        obj_instance->set_operation(M2MBase::GET_PUT_POST_ALLOWED);
+                                        coap_response = obj_instance->handle_post_request(_nsdl_handle,coap_header,this);
+                                    }
```
Comment (anttiylitokola) [inline source/m2mnsdlinterface.cpp]: trailing space

**Comment by yogpan01** [comment]
Cancelling this PR as changes are already merged last week.

**Pull request closed without merging**